README.md

```diff
@@ -170,6 +170,28 @@ class PostSerializer < ActiveModel::Serializer
 end
 ```
 
+### Context-aware predicate methods
+
+If you want to serialize attributes or associations differently depending on the context they're serialized, you can use:
+
+```ruby
+class PostSerializer < ActiveModel::Serializer
+  attributes :id, :body
+
+  has_many :comments
+
+  def body
+    if listed? or associated? # same as `if nested?`
+      object.body.lines.first + '...'
+    else
+      object.body
+    end
+  end
+end
+```
```

Comment:
So, in the example here, when you're serializing a post, you want to use the body attribute, but when serializing a blog, you want the associated posts to use only the first line of the post body?

To me, that sounds like a scenario where I'd do something like

```ruby
class PostSerializer < ActiveModel::Serializer
 attributes :body
end
class BlogSerializer < ActiveModel::Serializer
  has_many :posts, serializer: BlogPostSerializer
end
class BlogPostSerializer < ActiveModel::Serializer
  def body
    super.lines.first
  end
end
```

Was that an approach you considered?